query/physicalplan/project.go

```diff
@@ -380,6 +498,96 @@ func (p plainProjection) Project(_ memory.Allocator, ar arrow.Record) ([]arrow.F
 	return nil, nil, nil
 }
 
+type convertProjection struct {
+	p    columnProjection
+	expr *logicalplan.ConvertExpr
+}
+
+func (p convertProjection) Name() string {
+	return p.expr.Name()
+}
+
+func (p convertProjection) String() string {
+	return p.expr.Name()
+}
+
+func (p convertProjection) convert(mem memory.Allocator, c arrow.Array) (arrow.Array, error) {
+	switch c := c.(type) {
+	case *array.Int64:
+		switch p.expr.Type {
+		case arrow.PrimitiveTypes.Float64:
+			return convertInt64ToFloat64(mem, c), nil
+		default:
+			return nil, fmt.Errorf("unsupported conversion from %s to %s", c.DataType(), p.expr.Type)
+		}
+	default:
+		return nil, fmt.Errorf("unsupported conversion from %s to %s", c.DataType(), p.expr.Type)
+	}
+}
+
+func convertInt64ToFloat64(mem memory.Allocator, c *array.Int64) *array.Float64 {
+	defer c.Release()
```

Comment:
nit: the caller should release this

Reply:
> nit: the caller should release this

need release, example:  select value * timestamp * 2 , floatvalue * convert(timestamp * 2, float64)

because timestamp * 2 is new array from binaryExprProjection, caller can't release it, so convertInt64ToFloat64 need to release it.

Reply:
Not sure I understand. This array passed into `convertInt64ToFloat64` is retained somewhere (I believe in the call to `Project` in `convertProjection.Project`. I'm saying that the ownership for that array belongs to the `convertProjection` operator, so it should `Release` it in a defer (i.e. here: https://github.com/polarsignals/frostdb/pull/702/files/8296f351dfd060c9dbb4bc055e6f9d4b054ea184#diff-9cc015062538372988d7c41909b5a31a465ffe061d4d70bf5f24e41fdc8044afR543)

Reply:
I agree, also prevents us from making mistakes in the "lower" functions.